packages/app-builder-lib/src/node-module-collector/index.ts

```diff
@@ -3,14 +3,34 @@ import { PnpmNodeModulesCollector } from "./pnpmNodeModulesCollector"
 import { YarnNodeModulesCollector } from "./yarnNodeModulesCollector"
 import { detect, PM, getPackageManagerVersion } from "./packageManager"
 import { NodeModuleInfo } from "./types"
+import { exec } from "builder-util"
+
+async function isPnpmProjectHoisted(rootDir: string) {
+  const command = process.platform === "win32" ? "pnpm.cmd" : "pnpm"
+  const config = await exec(command, ["config", "list"], { cwd: rootDir, shell: true })
+  const lines = config
+    .split("\n")
+    .map(line => line.split("="))
+    .reduce<Record<string, string>>((accum, curr) => {
+      const [key, value] = curr
+      return {
+        ...accum,
+        [key]: value,
```

Comment:
We'd better to perform a `trim` operation on both the beginning and end of the string, as some operating systems may introduce extra spaces.

Reply:
Good callout, updated!